std/range/package.d

```diff
@@ -11844,3 +11848,67 @@ pure @safe unittest
     static immutable r2 = [1, 2, 3, 4, 0, 0];
     assert(r1.padRight(0, 6).equal(r2));
 }
+
+private struct TakeWhile(alias pred, R)
+if (isInputRange!R && is(typeof(!unaryFun!pred(ElementType!R.init)) == bool))
+{
+    private R _range;
+
+    // Range primitives
+    bool empty() @property
+    {
+        return _range.empty || !(unaryFun!pred(_range.front));
+    }
+
+    auto front() @property
+    {
+        assert(!empty, "Attempting to fetch the front of an empty TakeWhile range");
+        return _range.front;
+    }
+
+    auto save() @property
+    {
+        return this;
+    }
+
+    void popFront() @property
+    {
+        assert(!empty, "Attempting to pop the front of an empty TakeWhile range");
+        _range.popFront;
+    }
+}
+
+/**
+Lazily produce elements of a range as long as the predicate holds
+for the produced elements.
+
+Params:
+    range = the range to take from while the predicate stands
+    pred = the prediacate used to take the elements
+
+Returns:
+    A `ForwardRange` which contains the taken elements.
+
+See_Also: $(LREF chain) to chain ranges
+ */
+auto takeWhile(alias pred, R)(R range) if (isInputRange!R)
```

Comment:
@CyberShadow `until`'s documentation actually already mentions `takeWhile`, saying

    This is similar to `takeWhile` in other languages.

So, as far as searchability goes, if they search the documentation, they'll find it. It just won't be in the links, because it's not actually a symbol, just a comment in the documentation.